File: SecondYear/praveer7398_PraveerNandan_2226ce1011_2/Week_3/selection_sort.java
import java.util.Scanner;

public class selection_sort {
    public static void selectionSort(int[] arr) {
        int n = arr.length;

        for (int i = 0; i < n - 1; i++) {
            int minIndex = i;

            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }

            
            int temp = arr[minIndex];
            arr[minIndex] = arr[i];
            arr[i] = temp;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        
        int n = scanner.nextInt();
        int[] arr = new int[n];

        
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        

        selectionSort(arr);

        
        for (int num : arr) {
            System.out.print(num + " ");
        }

        scanner.close();
    }
}

File: SecondYear/Arystar01_AryanRastogi_2226cse1169_2/Week_3/Selectionsort.c
#include<stdio.h>
void swap(int *x, int*y){
    int temp;
    temp=*x;
    *x=*y;
    *y=temp;
}
void selectionSort(int arr[], int n){
    int min_index;
    for(int i=0;i<n-1;i++){
        min_index=i;
        for(int j=i+1;j<n;j++){

            if(arr[j]<arr[min_index]){
                min_index=j;
            }
                swap(&arr[i],&arr[min_index]);
        }
    }
}
void printArray(int arr[],int n){
    for(int i=0;i<n;i++){
       
        printf("%d ",arr[i]);
    }
}
int main(){
    int arr[10],n,item;
    printf("Enter the number of terms: \n");
    scanf("%d",&n);
    for(int i=0;i<n;i++){
        printf("Enter the number:");
        scanf("%d",&arr[i]);
    }
    selectionSort(arr,n);
    printArray(arr,n);
    return 0;
}